components/textfield/stories/template.js

```diff
@@ -148,3 +173,130 @@ export const Template = ({
+				<div
+					style=${styleMap({
+						marginBottom: "10px",
+					})}
+				>
+					${Typography({
+						semantics: "detail",
+						size: "l",
+						content: [item.testHeading]
+					})}
+				</div>
+			` : ""}
+			${Template({...item})}
+		</div>
+	`)}
+`;
+
+const Variants = ({ testData, stateData }) => (args) => html`
+	<div style="display: flex; flex-direction: column; gap: 32px;">
+		${window.isChromatic() ?
+			html`
+				${renderDataWithHeading(testData)}
+				${renderDataWithHeading(stateData)}
+			`
+			: html`
+				${Template({...args})}
+				${Template({
+					...args,
+					multiline: true,
+				})}
+			`
+		}
+	</div>
+`;
+
+export const TextFieldGroup = Variants({
+	Template,
+	testData: [{
+	},
+	{
+		testHeading: "With field label",
+		displayLabel: true,
+		labelText: "Username",
+	},
+	{
+		testHeading: "With side label",
+		displayLabel: true,
+		labelText: "Username",
+		labelPosition: "side",
+	},
+	{
+		testHeading: "With value",
+		displayLabel: true,
+		labelText: "Username",
+		value: "UsernameWiderThanInput@ReallyLongEmail.com"
+	},
+	{
+		testHeading: "With placeholder",
+		placeholder: "Username",
+	},
+	{
+		testHeading: "With help text",
+		displayLabel: true,
+		labelText: "Username",
+		helpText: "Please enter a username between 8 and 16 characters",
+	},
+	{
+		testHeading: "Text area",
+		multiline: true,
+	},
+	{
+		testHeading: "Text area with label",
+		displayLabel: true,
+		labelText: "Username",
+		multiline: true,
+	},
+	{
+		testHeading: "Text area with value",
+		displayLabel: true,
+		labelText: "Username",
+		multiline: true,
+		value: "Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo. Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt.",
+	},
+	{
+		testHeading: "Text area with character count",
+		displayLabel: true,
+		labelText: "Comments",
+		multiline: true,
+		hasCharacterCount: true,
+		characterCount: 400,
+		value: "Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo. Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt.",
+	}],
+	stateData: [{
+		testHeading: "Invalid",
+		isInvalid: true,
+	},
+	{
+		testHeading: "Valid",
+		isValid: true,
+	},
+	{
+		testHeading: "Focused",
+		isFocused: true,
+	},
+	{
+		testHeading: "Keyboard focused",
+		isKeyboardFocused: true,
+	},
+	{
+		testHeading: "Disabled",
+		isDisabled: true,
+	},
+	{
+		testHeading: "Required",
+		isRequired: true,
+	},
+	{
+		testHeading: "Read-only",
+		value: "username@reallylongemail.com",
```

Comment:
Does this need to be changed to the new "UsernameWiderThanInput@ReallyLongEmail.com" at all? It's not really necessary I guess, but maybe it would cut down on VRT diffs?
<img width="274" alt="Screenshot 2024-07-25 at 12 19 14 PM" src="https://github.com/user-attachments/assets/5c0ee666-4cde-4ef2-8631-25bfa08327be">

Reply:
I'll change it! It doesn't look like `main` is applying a value at all, which I must have disagreed with, because I added one.